pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -100,24 +110,33 @@ type desiredStateOfWorldPopulator struct {
 	kubeContainerRuntime      kubecontainer.Runtime
 	timeOfLastGetPodStatus    time.Time
 	keepTerminatedPodVolumes  bool
+	hasAddedPods              bool
 }
 
 type processedPods struct {
 	processedPods map[volumetypes.UniquePodName]bool
 	sync.RWMutex
 }
 
-func (dswp *desiredStateOfWorldPopulator) Run(stopCh <-chan struct{}) {
-	wait.Until(dswp.populatorLoopFunc(), dswp.loopSleepDuration, stopCh)
+func (dswp *desiredStateOfWorldPopulator) Run(sourcesReady config.SourcesReady, stopCh <-chan struct{}) {
+	wait.Until(dswp.populatorLoopFunc(sourcesReady), dswp.loopSleepDuration, stopCh)
 }
 
 func (dswp *desiredStateOfWorldPopulator) ReprocessPod(
 	podName volumetypes.UniquePodName) {
 	dswp.deleteProcessedPod(podName)
 }
 
-func (dswp *desiredStateOfWorldPopulator) populatorLoopFunc() func() {
+func (dswp *desiredStateOfWorldPopulator) HasAddedPods() bool {
+	return dswp.hasAddedPods
+}
+
+func (dswp *desiredStateOfWorldPopulator) populatorLoopFunc(sourcesReady config.SourcesReady) func() {
 	return func() {
+		if sourcesReady.AllReady() {
+			defer func() { dswp.hasAddedPods = true }()
+		}
+
```

Comment:
Are we using `defer` unnecessarily here? How about adding:

```
if sourcesReady.AllReady() {
			dswp.hasAddedPods = true
		}
```

line after `findAndAddNewPods` invocation?